app/assets/javascripts/discourse/components/desktop-notification-config.js.es6

```diff
@@ -44,27 +46,68 @@ export default Ember.Component.extend({
   },
 
   @computed("isGrantedPermission", "notificationsDisabled")
-  isEnabled(isGrantedPermission, notificationsDisabled) {
+  isEnabledDesktop(isGrantedPermission, notificationsDisabled) {
     return isGrantedPermission ? !notificationsDisabled : false;
   },
 
-  actions: {
-    requestPermission() {
-      Notification.requestPermission(() => this.propertyDidChange('notificationsPermission'));
+  @computed
+  isEnabledPush: {
+    set(value) {
+      const user = this.currentUser;
+      if(!user) {
+        return false;
+      }
+      pushNotificationKeyValueStore.setItem(pushNotificationUserSubscriptionKey(user), value);
+      return pushNotificationKeyValueStore.getItem(pushNotificationUserSubscriptionKey(user));
     },
+    get() {
+      const user = this.currentUser;
+      return user ? pushNotificationKeyValueStore.getItem(pushNotificationUserSubscriptionKey(user)) : false;
+    }
+  },
 
+  @computed("isEnabledDesktop", "isEnabledPush")
+  isEnabled(isEnabledDesktop, isEnabledPush) {
```

Comment:
Minor but you might want to try `Ember.computed.or` here instead. One line instead of 4.